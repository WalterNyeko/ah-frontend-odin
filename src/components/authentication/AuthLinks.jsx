import React from 'react';
import PropTypes from 'prop-types';
import defaultUserImage from 'user.png';

const AuthLinks = ({ user }) => (
  <React.Fragment>
    <li className="nav-item d-flex align-items-center justify-content-between">
      <div className="bg-white  rounded rounded-circle">
        <img src={defaultUserImage} alt={user.username} className="user-avatar" />
      </div>
      <h4 className="text-white ml-2">{user.username}</h4>
    </li>
    <li className="nav-item">
      <button className="nav-link text-light btn-link">Logout</button>
    </li>
  </React.Fragment>
);

AuthLinks.propTypes = {
  user: PropTypes.object.isRequired,
};

export default AuthLinks;
