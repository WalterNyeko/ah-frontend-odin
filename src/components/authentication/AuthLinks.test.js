import React from 'react';
import renderer from 'react-test-renderer';
import AuthLinks from './AuthLinks';

describe('AuthLinks', () => {
  test('It renders the same UI', () => {
    const user = { username: 'roland', email: 'rolandmbasa@gmail.com' };

    expect(renderer.create(<AuthLinks user={user} />).toJSON()).toMatchSnapshot();
  });
});
