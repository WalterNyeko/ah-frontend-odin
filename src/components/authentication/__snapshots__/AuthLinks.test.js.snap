// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLinks It renders the same UI 1`] = `
Array [
  <li
    className="nav-item d-flex align-items-center justify-content-between"
  >
    <div
      className="bg-white  rounded rounded-circle"
    >
      <img
        alt="roland"
        className="user-avatar"
        src="user.png"
      />
    </div>
    <h4
      className="text-white ml-2"
    >
      roland
    </h4>
  </li>,
  <li
    className="nav-item"
  >
    <button
      className="nav-link text-light btn-link"
      onClick={[MockFunction]}
    >
      Logout
    </button>
  </li>,
]
`;
